### PR TITLE
Added Linear Search [Golang]

### PR DESCRIPTION
**Pull request by singhpratyush**
#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [ ] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous timespan of 7 days.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,37 +10,37 @@ Community (college) maintained list of Algorithms and Data Structures implementa
...
+| [Sieve of Eratosthenes](https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes) | [\[X\]](sieve_of_eratosthenes/sieveOfEratosthenes.c) | | | [\[X\]](sieve_of_eratosthenes/sieve_of_eratosthenes.py) | |
 
 
 ## Implemented Data Structures
 
-| Data Structure | C | CPP | Java | Python |
-|:--------------|:----------------:|:----------------:|:----------------:|:-----------------:|
+| Data Structure | C | CPP | Java | Python | Golang |  
+|:--------------|:----------------:|:----------------:|:----------------:|:-----------------:|:-----------------:|
 | [Binary Search Tree](https://en.wikipedia.org/wiki/Binary_search_tree) | | | [\[X\]](binary_search_tree/BinarySearchTree.java) | |
```
Comment (aviaryan) [inline README.md]: Append " |" here.

File: linear_search/linear-search.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    return -1
+}
+
+func main() {    
+    array := []int{0, 2, 4, 6, 8, 10, 12, 14, 16, 18}
+    index := linearSearch(array, 10)
+    if index == -1 {
+        fmt.Println("Number not found")
+    } else {
+        fmt.Println("10 found at index ", index)
```
Comment (aviaryan) [inline linear_search/linear-search.go]: No need for space after "index" as different parameters in `fmt.Print` are already separated by a space when printed.

File: README.md [hunk 3]
```diff
@@ -10,37 +10,37 @@ Community (college) maintained list of Algorithms and Data Structures implementa
...
-| [Modular Exponential](http://www.geeksforgeeks.org/modular-exponentiation-power-in-modular-arithmetic/) | | | [\[X\]](modular_exponential/ModularExponential.java) | [\[X\]](modular_exponential/modular_exponential.py) |
-| [N-Queen Problem](https://en.wikipedia.org/wiki/Eight_queens_puzzle) | | [\[X\]](n_queen_problem/NQueenProblem.cpp) | | | |
-| [Prime Factor](https://en.wikipedia.org/wiki/Prime_factor) | [\[X\]](prime_factor/prime_factor.c) | | | |
-| [Prims](https://en.wikipedia.org/wiki/Prim%27s_algorithm) | | | [\[X\]](prims/Prims.java) | |
-| [Quick Select](https://en.wikipedia.org/wiki/Quickselect) | [\[X\]](quick_select/quick_select.c) | | | |
-| [Quicksort](https://en.wikipedia.org/wiki/Quicksort) | [\[X\]](quicksort/quicksort.c) | | [\[X\]](quicksort/QuickSort.java) | [\[X\]](quicksort/quick_sort.py) |
-| [Sieve of Eratosthenes](https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes) | [\[X\]](sieve_of_eratosthenes/sieveOfEratosthenes.c) | | | [\[X\]](sieve_of_eratosthenes/sieve_of_eratosthenes.py) |
+| Algorithm | C | CPP | Java | Python | Golang |
```
Comment (aviaryan) [inline README.md]: Trailing space after `| Golang |`.

Comment (singhpratyush) [inline README.md]: Need to add that package control plugin I guess.

Comment (singhpratyush) [inline README.md]: What's the name btw?

Comment (aviaryan) [inline README.md]: Use GoSublime. It does that + makes your sublime text nothing less than an IDE for Golang. **Very highly recommended.**

**Pull request merged**